### PR TITLE
Add a build step, in order to create variants of the add-on programatically

### PR DESCRIPTION
**Pull request by mythmon**
I removed the watch command from `package.json` because I didn't want to create the infrastructure to watch files and rebuild as needed. If that becomes needed in the long run, I'll just switch to Gulp.

CC @rhelmer

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,10 +1,17 @@
 # Normandy NextGen Study Example
 
-Demonstrates how to make a 2019-style Normandy add-on study
+Demonstrates how to make a 2019-style Normandy add-on study.
+
+There are three _variants_ of the add-on provided: "a", "b", and "c". Each is
+identical except for it's add-on ID. This is to test different study branches
```
Comment (rhelmer) [inline README.md]: "its" is the possessive form of it, https://www.dictionary.com/e/its-vs-its/

File: README.md [hunk 2]
```diff
@@ -1,10 +1,17 @@
...
+having different XPIs. The add-on ID is shown in the UI of the add-on, in
+order to distinguish different XPIs.
 
 ## Usage of the add-on
 
-When installed correctly, they add-on places a green puzzle piece browser action in the top right of the browser. Clicking this icon will open a page with details about the add-on's operation. It should show several things.
+When installed correctly, they add-on places a green puzzle piece browser
```
Comment (rhelmer) [inline README.md]: `s/they add-on/the add-on/`

File: README.md [hunk 3]
```diff
@@ -73,9 +80,12 @@ await AddonStudies.add({
...
 
-```
-yarn install --frozen-lockfile
+```bash
+yarn install
 yarn build
 ```
 
-A built add-on will be placed in `./web-ext-artifacts/`.
+Several built add-on will be placed in `./web-ext-artifacts/`. Each is nearly
```
Comment (rhelmer) [inline README.md]: `s/add-on/add-ons/`

File: build.js [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+                const newPath = targetPath.replace(/\.tmpl$/, "");
+                const content = await fsp.readFile(path, { encoding: "utf8" });
+                const rendered = template(content, templateData);
+                await fsp.writeFile(newPath, rendered);
+              } else {
+                await fsp.copyFile(path, targetPath);
+              }
+            } else {
+              throw new Error(
+                `Can't handle path at ${path} - neither file nor directory`
```
Comment (rhelmer) [inline build.js:68]: Nice

**Review by rhelmer — APPROVED** [review]

**Review by rehandalal — APPROVED** [review]

**Pull request merged**